### ISSUE TITLE
`#[doc = format!(...)]` silently fails to produce output instead of producing compilation error

### ISSUE THREAD

**Issue by JohnScience** [rust-lang/rust#99377]
In order to keep the doc test for [evcxr](https://github.com/google/evcxr) fresh, I wanted to use

```rust
/// Blah-blah-blah
#[doc = format!(":dep nalgebra_latex = {{ version = \"{}\", features = [\"lin_sys\", \"evcxr\"] }}", env!("CARGO_PKG_VERSION"))]
/// blah-blah-blah
```

I expected to see a compile error at least when I would run `cargo doc`.

Instead, it silently produced an empty line.

### Meta

`rustc --version --verbose`:
```
rustc 1.64.0-nightly (1c7b36d4d 2022-07-12)
binary: rustc
commit-hash: 1c7b36d4db582cb47513a6c7176baaec1c3346ab
commit-date: 2022-07-12
host: x86_64-pc-windows-msvc
release: 1.64.0-nightly
LLVM version: 14.0.6
```

<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

**Comment by GuillaumeGomez** [comment]
I guess not all macros should be allowed in attributes. cc @petrochenkov (I think you were the one who implemented this feature?)

**Comment by petrochenkov** [comment]
If the output of a macro is not a string literal, then it should produce an error.
Not sure why the error is not produced in this case, someone (probably not me) needs to investigate.

**Comment by LoganDark** [comment]
FWIW, `concat!` does work in `#[doc]` attributes, so that is the preferred way to do this. I do agree this should be a compile error though.

**Comment by JohnScience** [comment]
I cannot reproduce it. So I guess it's worth closing.
